CMakeLists.txt

```diff
@@ -212,6 +482,31 @@ if (dht)
 	endif()
 endif()
 
+list(APPEND sources2 ${include_files})
+
+if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
+	add_compile_options(
+		-Weverything
+		-Wno-documentation
+		-Wno-c++98-compat-pedantic
+		-Wno-padded
+		-Wno-global-constructors
+		-Wno-exit-time-destructors
+		-Wno-weak-vtables)
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
+	add_compile_options(
+		-Wall
+		-Wextra
+		-Wpedantic
+		-Wparentheses
+		-Wvla
+		-Wc++11-compat
+		-Wno-format-zero-length
+		-ftemplate-depth=512)
+endif()
+
```

Comment:
These should probably use `target_compile_options`.

Reply:
Looking at this again I realize that these options shouldn't be propagated to dependents, but we still want them to apply to examples/tests/etc. So you had it right the first time, `add_compile_options` was correct here.

Reply:
right, but before reverting, what do you think of `-ftemplate-depth=512`? (to propagate, or not)

Reply:
I don't think it should. We're setting it lower than the default specified by C++11 (1024) to catch excessive template recursion and that shouldn't be imposed on dependents.